components/authz-service/storage/postgres/migration/sql/49_add_status_field_to_rules.up.sql

```diff
@@ -0,0 +1,154 @@
+BEGIN;
+
+CREATE OR REPLACE FUNCTION
+  query_staged_and_applied_rules(_project_filter TEXT[])
```

Comment:
does this query concatenate the rows from the two tables? 

for `GetStagedOrAppliedRule(id)` we return the staged rule if it exists, otherwise return the applied rule (due to an acceptance requirement that users should always see the latest staged change). we may want to make this query work the same way - returning only one version of each rule. 

although since i don't know of a specific case where we're using this query yet, this is probably ok for now.

Reply:
A couple points.
1. Looking at `GetStagedOrAppliedRule` I do not believe it does what you think, @bcmdarroch . Looks like the SQL underlying that call, `query_staged_rule`, works exactly the same as this new function, returning both staged and applied rules.
2. I do take issue with the name of this new function `query_staged_and_applied_rules` compared to the aforementioned function `query_staged_rule`. Please rename one or the other so they differ only in plurality. 😁

Reply:
I think we've resolved that in standup:
> returning both staged and applied rules.

but we use `QueryRowContext`, so we're basically returning the staged one first (if it exists), so we're OK here, even if by accident. There might be a cleaner solution, but this one will do. 😉

Reply:
`query_staged_rules` checks both `applied` and `staged` but [only returns the applied rule if no staged version exists](https://github.com/chef/automate/blob/master/components/authz-service/storage/postgres/migration/sql/47_staged_rules_tables.up.sql#L57).